src/test/interface/io/json/json_data_test.cpp

```diff
@@ -52,6 +54,27 @@ void test_real_var(cmdstan::json::json_data &jdata, const std::string &text,
     EXPECT_EQ(expected_vals[i], vals[i]);
 }
 
+void test_complex_var(cmdstan::json::json_data &jdata, const std::string &text,
+                      const std::string &name,
+                      const std::vector<std::complex<double>> &expected_vals,
+                      const std::vector<size_t> &expected_dims) {
+  EXPECT_EQ(true, (jdata.contains_r(name) || jdata.contains_i(name)));
+  std::vector<size_t> dims = jdata.dims_r(name);
+  if (dims.size() == 1) {
```

Comment:
You don't need this first conditional.  `pop_back()` will be equivalent to `clear()` if there is only a single element in the container.  You generally don't want to branch for this kind of thing as branching is very inefficient when there's branch-point misprediction.